### ISSUE TITLE
"Ignore this notification" isn't working

### ISSUE THREAD

**Issue by sigismund** [magento/magento2#11365]
### Preconditions

1. Magento 2.2
2. 3 store views

### Steps to reproduce

1. Set-up wrong tax settings
2. Message similar "To apply the discount on prices including tax and ..." will be display as system message.
3. Click on "ignore this notification"

### Expected result

1. System message (ignoreTaxNotification) shouldn't be displayed in future because it's supposed to be ignored

### Actual result

1. Message is displayed regardless of being on ignore list.

##### Probably unrelated

This is most likely unrelated, but even after changing values to "correct" mode all checks in function `checkSettings` in `ApplyDiscountOnPrices` are still returning false even though they are all set to true.

Checks I mentioned:
```
$this->taxConfig->priceIncludesTax($store)
!$this->taxConfig->applyTaxAfterDiscount($store)
!$this->taxConfig->discountTax($store);
```

**Comment by magento-engcom-team** [comment]
@sigismund, thank you for your report.
We've created internal ticket(s) MAGETWO-81606 to track progress on the issue.

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-82753

**Comment by okorshenko** [comment]
The issue has been fixed and delivered to 2.2-develop branch. Will be available with 2.2.2 release

**Comment by magento-engcom-team** [comment]
Hi @sigismund. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1285 by @magento-engcom-team in 2.3-develop branch
Related commit(s):
 - ad6c0af7dddbe2855a336acd3fd55994851e6d26
 - 6d3d70a72f08cba36c3aef6b59eea8db40eb4071
 - 6b0c02c5fd00d7a4237c4e1fae40dc265e3e43c3
 - c8abb139f8006826989fe4231dabc21fd4a02fd0
 - c6a9b5f7b13554958b547f387637c4288eb60b0b
 - c867b81af936e01a3d8ba8baa7128de06a5b7cd5
 - 88a3ad00d3ca97da847531c86bf64055dc1d790b
 - d87e316a2d6ee55debe2e712359adba8cb70fdb2
 - c76b04c9acde117c1021270500c8ded6954f6f05
 - 5ae21099aaa19699525c798f9ac0ef10db47b311
 - 931133abf574f1db96450f11f5980ef359d32967
 - 4c927a0585f11921c5e422b7e4a15b787e6338bd
 - 060a8f9f2b08ed97b31667eb76cace1dc0e642be
 - 4da2a7d2657b0da67f8be43bffc85f04b5b7144e
 - 008ef5583892d73c04628f003a2ce3d97fe0ed72
 - d3642e88d26a9240a1a4d648d37d984a785672a9
 - d37ce089d56c230f5896d5d7e922bd928b3726e6
 - b22b232563359f9cbcbf09709e63a87be7e81446
 - 6c02a2d3a272e872d549881d91ed6d5235a33f77
 - f2ec2ea175469636acec0bf9f65f0601854f89a2

The fix will be available with the upcoming patch release.
